war/src/main/java/io/lumeer/engine/rest/SecurityService.java

```diff
@@ -0,0 +1,410 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.lumeer.engine.rest;
+
+import io.lumeer.engine.api.LumeerConst;
+import io.lumeer.engine.api.dto.Role;
+import io.lumeer.engine.controller.OrganizationFacade;
+import io.lumeer.engine.controller.SecurityFacade;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.zip.DataFormatException;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+/**
+ * @author <a href="mailto:martin.glova@lumeer.io">Martin Glova</a>
+ */
+@Path("/roles/organizations/{organization}")
+@RequestScoped
+public class SecurityService implements Serializable {
+
+   @PathParam("organization")
+   private String organizationCode;
+
+   @Inject
+   private SecurityFacade securityFacade;
+   @Inject
+   private OrganizationFacade organizationFacade;
+
+   @PostConstruct
+   public void init() {
+      if (organizationCode == null)
+         throw new BadRequestException();
+
+      organizationFacade.setOrganizationCode(organizationCode);
+   }
+
+   private boolean isRoleCorrect(String resource, String roleName) {
+      return LumeerConst.Security.RESOURCE_ROLES.get(resource).contains(roleName);
+   }
+
+   /**
+    * Add a role for users and groups for an organization.
+    *
+    * @param role
+    *       the role to be added
+    * @param users
+    *       the users to be added the role for
+    * @param groups
+    *       the groups to be added the role for
+    */
+   @PUT
+   @Path("/roles/{role}")
+   public void addOrganizationUsersGroupsRole(
+         final @PathParam("role") String role,
+         final @QueryParam("users") List<String> users,
+         final @QueryParam("groups") List<String> groups) {
+      if (role == null || !isRoleCorrect(LumeerConst.Security.ORGANIZATION_RESOURCE, role))
+         throw new BadRequestException();
+
+      for (String user: users) {
+         securityFacade.addOrganizationUserRole(organizationCode, user, role);
+      }
+
+      for (String group: groups) {
+         securityFacade.addOrganizationGroupRole(organizationCode, group, role);
```

Comment:
This is highly inefficient since for every user and group we make a new database call. Consider adding appropriate method to the facade to handle such operations.